### PR TITLE
API: Change Period('NAT') to return NaT

### LINKED ISSUE #12759: API: Change Period('NAT') to return NaT, not Period instance
#### Code Sample, a copy-pastable example if possible

``Timedelta`` returns ``NaT`` when input can be regarded as ``NaT``. Otherwise ``Period`` returns its special representation (This was done in #7485 prior to ``Timedelta``).

```
pd.Timedelta('NaT')
# NaT

pd.Period('NaT', freq='M')
# Period('NaT', 'M')
```

#### Expected Output

```
pd.Period('NaT', freq='M')
# NaT
```

The fix should affects:

- [ ] ``Period`` and ``PeriodIndex``, ``Series`` ops
  - [ ] add, sub, comp
- [x] ``PeriodIndex`` creation from list-like which contains ``Period`` and ``NaT`` (#13430)
- [ ] ``PeriodIndex`` boxing
- [ ] ``.to_period``
- [ ]  ``__contains__`` any NaT-like(``pd.NaT, None, float('nan'), np.nan``)  (#13582)

#### output of ``pd.show_versions()``

Current master.

### PR DESCRIPTION
**Pull request by sinhrks**
- [x] closes #12759, #13582
- [x] tests added / passed
- [x] passes `git diff upstream/master | flake8 --diff`
- [x] whatsnew entry
- `pd.Period('NaT', freq='D')` now returns `pd.NaT` rather than its own nat repr. 
- To be compat with `Period` arithmetic, `pd.NaT` now supports addition / subtraction with `int` (previously `ValueError`).

```
pd.NaT + 1
# pd.NaT
```

Added tests for following:
- [x] `Period` and `PeriodIndex`, `Series` ops
  - [x] add, sub, comp
- [x] `PeriodIndex` creation from list-like which contains `Period` and `NaT` (#13430)
- [x] `PeriodIndex` boxing
- [x] `.to_period`
- [x]  `__contains__` any NaT-like(`pd.NaT, None, float('nan'), np.nan`)

### CODE REVIEW

**Comment by sinhrks** [comment]
CC @MaximilianR

File: doc/source/whatsnew/v0.19.0.txt [hunk 1]
```diff
@@ -425,6 +425,45 @@ Furthermore:
 - Passing duplicated ``percentiles`` will now raise a ``ValueError``.
 - Bug in ``.describe()`` on a DataFrame with a mixed-dtype column index, which would previously raise a ``TypeError`` (:issue:`13288`)
 
+.. _whatsnew_0190.api.periodnat:
+
+``Period('NaT')`` now returns ``pd.NaT``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Previously, ``Period`` has its own ``Periond('NaT')`` representation different from ``pd.NaT``. Now ``Period('NaT')`` has been changed to return ``pd.NaT``. (:issue:`12759`)
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v0.19.0.txt]: Periond -> Period

**Comment by jorisvandenbossche** [comment]
@sinhrks Can you rebase this? Then it will be easier to review the additional changes (as I think the first commit is now already in master?)

**Comment by sinhrks** [comment]
@jorisvandenbossche sure. updated.

**Comment by jreback** [comment]
thanks @sinhrks this is really nice!

**Pull request closed without merging**